Lib/test/support/__init__.py

```diff
@@ -1900,21 +1899,48 @@ def _run_suite(suite):
         raise TestFailed(err)
 
 
-def _match_test(test):
-    global match_tests
+class _MatchTests:
```

Comment:
I think the class is not needed. Just use a function which returns a function.

Reply:
Ok, done.